design-proposals/volume-claim-templates.md

```diff
@@ -0,0 +1,117 @@
+Along major benefits of using this populators we loss one of DataVolumes greatest achievements - generating a PVC with the optimal properties fitting the user's storage.  
+This design should help us gain the best of all worlds.
+
+## Motivation
+Harnessing k8s and CDI new volume populators API along with our knowledge of storage capabilities will allow the user to enjoy synchronization with k8s and better compatibility with backup and restore solutions as well as the smart creation of PVCs that best suits the user's storage.
+
+## Goals
+- Provide a method for creating VMs with PVCs that suits best the user's storage while using CDI volume populators.
+
+## Non Goals
+
+## User Stories
+* As a user creating VMs, I want to be able to create a PVC within the VM spec while getting the best PVC spec that is also compatible with backup and restore solutions.
+
+## Repos
+* **CDI**: Provide API to that receives a PVC spec and in case of missing properties such as VolumeMode and AccessModes fills the optimal values according to the storage class.
+* **CDI**: Update DataImportCron API to create/update VolumeCloneSources that can be consumed as sources for PVC handled by CDI volume populators. VolumeCloneSource will need to be able to be updated with different PVC source.
+* **Kubevirt**: Provide API to create a VM with minimal volume information and receive a VM with best suited populated PVC.
+* **Kubevirt**: Provide virtctl command to generate and create PVCs.
+* **SSP**: Manage setting up VolumeCloneSources for commonly supported operating systems updated by DataImportCrons.
+* **Common Templates**: Transition to use VolumeClaimTemplates in VM specs and use CDI's VolumeCloneSource as the source in the template.
+* **OCP UI**: Provide API to create a CDI Volume populator CR that can be used by the volumeClaim. In case of golden image source need to ensure VolumeCloneSource is linked properly to a PVC in order to determine whether the boot image for a common template is available.
+
+# Design
+
+A new field will be added to the VM object called **VolumeClaimTemplates**. This field will provide the user a way to define and create a PVC owned by the VM while providing partial/full PVC spec embedded in the VM spec. In case of partial spec we will make sure the preferred VolumeMode and AccessModes combination according to the StorageProfile associated with provided/default StorageClass are used.  
+This field should replace the DataVolumeTemplates field. Instead of providing a template for a DataVolume which eventually creates one, we will provide a PVC template and create it. DataVolumes and DataVolumeTemplates should be deprecated eventually.  
```

Comment:
Please do not say "replace" here as we cannot remove the dataVolumeTemplates field right now since many people depend on it.  "This field will complement the DataVolumeTemplates field."

Reply:
Are we planning on deprecating `DataVolumeTemplates` somehow in `v1` so we could remove it in `v2`?